msticpy/data/core/data_providers.py

```diff
@@ -449,22 +453,39 @@ def exec_query(self, query: str, **kwargs) -> Union[pd.DataFrame, Any]:
         """
         query_options = kwargs.pop("query_options", {}) or kwargs
         query_source = kwargs.pop("query_source", None)
-        result = self._query_provider.query(
-            query, query_source=query_source, **query_options
-        )
+
         if not self._additional_connections:
+            result = self._query_provider.query(
```

Comment:
Moved the 'single' query here to fix an issue where executing a query required a connect() even though multiple connections were added via the `add_connection` function

Reply:
Makes sense